docs/content/envoy-tutorial-gloo-edge.md

```diff
@@ -0,0 +1,268 @@
+
+## Steps
+
+### 1. Start Minikube
+
+```bash
+minikube start
+```
+
+### 2. Setup and Configure Gloo Edge
+
+```bash
+helm repo add gloo https://storage.googleapis.com/solo-public-helm
+helm upgrade --install --namespace gloo-system --create-namespace gloo gloo/gloo
+kubectl config set-context $(kubectl config current-context) --namespace=gloo-system
+```
+
+Ensure all the pods are running using `kubectl get pod` command.
+
+### 3. Create Virtual Service and Upstream
+
+[Virtual Services](https://docs.solo.io/gloo-edge/latest/introduction/architecture/concepts/#virtual-services) define a set of route rules, security configuration, rate limiting, transformations, and other core routing capabilities supported by Gloo Edge.
+
+[Upstreams](https://docs.solo.io/gloo-edge/latest/introduction/architecture/concepts/#upstreams) define destinations for routes.
+
+Save the configuration as **vs.yaml**.
+
+```yaml
+apiVersion: gloo.solo.io/v1
+kind: Upstream
+metadata:
+  name: httpbin
+spec:
+  static:
+    hosts:
+      - addr: httpbin.org
+        port: 80
+---
+apiVersion: gateway.solo.io/v1
+kind: VirtualService
+metadata:
+  name: httpbin
+spec:
+  virtualHost:
+    domains:
+      - '*'
+    routes:
+      - matchers:
+         - prefix: /
+        routeAction:
+          single:
+            upstream:
+              name: httpbin
+              namespace: gloo-system
+        options:
+          autoHostRewrite: true
+```
+
+```bash
+kubectl apply -f vs.yaml
+```
+
+### 4. Test Gloo
+
+For simplification port-forwarding will be used. Open another terminal and execute.
+
+```bash
+kubectl port-forward deployment/gateway-proxy 8080:8080
+```
+
+The `VirtualService` created in the previous step forwards requests to http://httpbin.org
+
+Let's test that Gloo works properly by running the below command in the first terminal.
+
+```bash
+curl -XGET -Is localhost:8080/get | head -n 1
+HTTP/1.1 200 OK
+
+curl http -XPOST -Is localhost:8080/post | head -n1
+HTTP/1.1 200 OK
+```
+
+### 5. Define an OPA Policy
+
+The following OPA policy only allows `GET` requests.
+
+**policy.rego**
+
+```rego
+package envoy.authz
+
+import input.attributes.request.http as http_request
+
+default allow = false
+
+allow {
+    action_allowed
+}
+
+action_allowed {
+  http_request.method == "GET"
+}
+```
+
+Next we build an OPA bundle.
+
+```bash
+opa build policy.rego
+```
+
+And now we serve the OPA bundle created above using Nginx.
+
+```bash
+docker run --rm --name bundle-server -d -p 8888:80 -v ${PWD}:/usr/share/nginx/html:ro nginx:latest
+```
+
+### 6. Setup OPA-Envoy
+
+Create a deployment as shown below and save it in **deployments.yaml**
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: opa
+  labels:
+    app: opa
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: opa
+  template:
+    metadata:
+      labels:
+        app: opa
+    spec:
+      containers:
+        - name: opa
+          image: openpolicyagent/opa:0.26.0-envoy
```

Comment:
Replace `openpolicyagent/opa:0.26.0-envoy` with `openpolicyagent/opa:{{< current_opa_envoy_docker_version >}}` to automatically pick up the latest version.

Reply:
Will change this to `openpolicyagent/opa:latest-envoy`? this seems to be what is used in the other tutorials.